crates/ruff_formatter/src/printer/mod.rs

```diff
@@ -1493,7 +1498,13 @@ impl<'a, 'print> FitsMeasurer<'a, 'print> {
                                     return Fits::No;
                                 }
                                 match args.measure_mode() {
-                                    MeasureMode::FirstLine => return Fits::Yes,
+                                    MeasureMode::FirstLine => {
+                                        return if exceeds_width(self, args) {
```

Comment:
This is a bug fix in the printer. So far, it always returned `Fits::Yes` if a text contained a newline character regardless if the text before the newline exceeded the line width. This was incorrect.